packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -613,6 +614,7 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
 
   /**
    * Whether Private DNS is supported by default.
+   * If the interface endpoint doesn't support Private DNS, privateDnsDefault will be set false.
```

Comment:
Private connections are supported, but private DNS is not supported for DynamoDB.

Document is [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/privatelink-interface-endpoints.html#privatelink-considerations).

> AWS PrivateLink for Amazon DynamoDB does not support the following:
> * Private and Hybrid Domain Name System (DNS) services

I tried enabling private DNS through the Management Console  and CloudFormation, but it resulted in an error.

![スクリーンショット 2024-10-02 9 04 25](https://github.com/user-attachments/assets/cfe17ef5-888c-49c7-983c-61f61c5368ad)
![スクリーンショット 2024-10-02 21 11 56](https://github.com/user-attachments/assets/abaab616-3211-488b-84a5-d08f164a06fe)

Previously, similar restrictions existed for S3 interface endpoint, but resolved through an update.

Ref:
* https://aws.amazon.com/about-aws/whats-new/2023/03/amazon-s3-private-connectivity-on-premises-networks/?nc1=h_ls

Currently, DynamoDB is the only service where private DNS cannot be used with interface VPC endpoints. 
This is likely due to the fact that DynamoDB originally only had Gateway VPC Endpoint.

I think we could remove the logic that sets privateDnsDefault to false specifically for DynamoDB.

In this case, if privateDnsDefault is not specified, it defaults to true and results in a deployment error. However, deployment can succeed if privateDnsDefault is explicitly set to false.
Furthermore, since DynamoDB may support private DNS in the future, the process of setting the default to false might become unnecessary.

What do you think?

Reply:
If it fails at deploy time, I think it should be fine that we don't need the extra logic on our end. The error message also seems descriptive enough that users could figure out the source of the issue and resolve it on their end.

So going forward, I think we can scrap the `getPrivateDnsDefault` method and the related documentation around it. It'd be nice to catch it earlier, but I don't think the added complexity is worth it, especially if this only applies to a single service.

Reply:
Thank you.
I have only added the DynamoDB interface endpoint and removed the methods.
I also added a unit test and an integ test for when privateDnsEnabled is set to false.